src/verbs/queue_pair.rs

```diff
@@ -356,3 +391,42 @@ impl QueuePairAttribute {
 }
 
 // TODO(zhp): trait for QueuePair
+
+#[derive(Debug)]
+pub struct PostSendGuardInner {
+    // TODO: add normal wr_list for traditional ibv_post_send
+    qp: Option<NonNull<ibv_qp>>,
+    qp_ex: Option<NonNull<ibv_qp_ex>>,
+}
+
+impl Drop for PostSendGuardInner {
+    fn drop(&mut self) {
+        match (self.qp, self.qp_ex) {
+            (None, Some(qp_ex)) => unsafe {
+                ibv_wr_complete(qp_ex.as_ptr());
```

Comment:
Get some problems here, both `ibv_post_send` and `ibv_wr_complete` would return error when user doesn't use the API correctly, so how can we tell user the problem, and let user decide what to do? Or we just panic and let user fix their code.

As the docs said,

```
Applications should use this API in a way that does not create failures. The individual APIs do not return a failure indication to avoid branching.

If a failure is detected during operation, for instance due to an invalid argument, then ibv_wr_complete() will return failure and the entire posting will be aborted.
```

User **should** use this API in a way that does not create failures. But I doubt it wouldn't always be possible, especially when we need to handle `ibv_post_send` failed.